lib/segment/src/index/hnsw_index/graph_layers.rs

```diff
@@ -301,7 +317,7 @@ impl GraphLayers {
     #[cfg(feature = "testing")]
     pub fn compress_ram(&mut self) {
         use crate::index::hnsw_index::graph_links::GraphLinksConverter;
-        assert!(!self.links.compressed() && !self.links.on_disk());
```

Comment:
Note: I've removed this check because the mmap+populate change breaks hnsw_search_graph.